internal/clusterconfig/aws_client.go

```diff
@@ -0,0 +1,31 @@
+package clusterconfig
+
+import (
+	"net/http"
+	"time"
+)
+
+const awsMetadataHost = "http://169.254.169.254"
```

Comment:
Only using this seems to be not reliable, since also other cloud provider use it, e.g. Azure: https://learn.microsoft.com/en-us/azure/virtual-machines/instance-metadata-service

Reply:
I'll add to the ip the exact http endpoint which will differentiate between providers. Eg /latest/meta-data will not be available outside of AWS.